src/python/espressomd/visualizationOpenGL.pyx

```diff
@@ -204,18 +326,18 @@ def drawSystemParticles(self):
             ptype = self.particles['types'][pid]
             ext_f = self.particles['ext_forces'][pid]
 
-            # Size: Lennard Jones Sigma,
-            if self.specs['particle_sizes'] == 'auto':
-                lj_sig = self.system.non_bonded_inter[ptype, ptype].lennard_jones.get_params()[
-                    'sigma']
-                radius = lj_sig / 2.0
-                if radius == 0:
-                    radius = self.sizeByType(ptype)
-
-            elif self.specs['particle_sizes'] == 'type':
-                radius = self.sizeByType(ptype)
+            radius = 0.5
+            try:
+                if callable(self.specs['particle_sizes']):
+                    radius = self.specs['particle_sizes'](ptype)
+                elif isinstance(self.specs['particle_sizes'], (list,tuple,np.ndarray)):
```

Comment:
You could `except TypeError`. If you don't like that, use `hasattr(obj, '__getitem__')` to detect whether an object is subscriptable instead of a try-except. Whichever you choose, please make the same change in the Mayavi visualizer.